Dockerfile

```diff
@@ -22,21 +22,24 @@ COPY homeassistant/package_constraints.txt homeassistant/homeassistant/
 RUN \
     uv pip install \
         --no-build \
+        --no-cache-dir \
```

Comment:
Nice! Well spotted; let's disable this on the docker base image level, just like we do for `pip`.

This is fine to do in a parallel PR, so we can get this out in the current release schedule for tomorrow.